test/AspNet.Security.OAuth.Providers.Tests/XiaoMi/XiaoMiTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.AspNetCore.WebUtilities;
+
+namespace AspNet.Security.OAuth.XiaoMi;
+
+public class XiaoMiTests : OAuthTests<XiaoMiAuthenticationOptions>
+{
+    public XiaoMiTests(ITestOutputHelper outputHelper)
+    {
+        OutputHelper = outputHelper;
+    }
+
+    public override string DefaultScheme => XiaoMiAuthenticationDefaults.AuthenticationScheme;
+
+    protected internal override void RegisterAuthentication(AuthenticationBuilder builder)
+    {
+        builder.AddXiaoMi(options => ConfigureDefaults(builder, options));
+    }
+
+    [Theory]
+    [InlineData(ClaimTypes.NameIdentifier, "my-union-id")]
+    [InlineData(ClaimTypes.Name, "John Smith")]
+    [InlineData("urn:xiaomi:miliaoIcon", "https://xiaomi.local/image.png")]
+    [InlineData("urn:xiaomi:unionId", "my-union-id")]
+    [InlineData("urn:xiaomi:miliaoNick", "John Smith")]
+    public async Task Can_Sign_In_Using_XiaoMi(string claimType, string claimValue)
+    {
+        // Arrange
+        using var server = CreateTestServer();
+
+        // Act
+        var claims = await AuthenticateUserAsync(server);
+
+        // Assert
+        AssertClaim(claims, claimType, claimValue);
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task BuildChallengeUrl_Generates_Correct_Url(bool usePkce)
```

Comment:
If `BuildChallengeUrl` doesn't need to be overridden, then this whole test can be deleted.